File: samples/python/AWS S3/main.py
import os
import json
import boto3
from botocore.exceptions import ClientError

from flask import Flask
from flask import request


app = Flask(__name__)


REGION_NAME = 'us-west-2'
BUCKET_NAME = 'my-sample-bucket'
FILE_NAME   = 'file1.json'


@app.route("/")
def index():
    return { 'status' : 'ok' }, 200


@app.route("/upload", methods=['POST'])
def upload():
    # Check that the post body is valid JSON
    if not request.is_json:
        return { 'error': 'Invalid JSON' }, 400

    # Parse the JSON into a Python dictionary
    data = request.get_json()

    AWS_ACCESS_KEY=os.environ['AWS_ACCESS_KEY']
    AWS_SECRET_KEY=os.environ['AWS_SECRET_KEY']

    # Write the JSON to S3
    s3 = boto3.client('s3', region_name=REGION_NAME, aws_access_key_id=AWS_ACCESS_KEY, aws_secret_access_key=AWS_SECRET_KEY)

    with open(FILE_NAME, 'w') as f:
        json.dump(data, f)

    s3.upload_file(FILE_NAME, BUCKET_NAME, FILE_NAME)

    return { 'status': 'ok' }, 200


@app.route("/download", methods=['GET'])
def download():
    AWS_ACCESS_KEY=os.environ['AWS_ACCESS_KEY']
    AWS_SECRET_KEY=os.environ['AWS_SECRET_KEY']

    # Write the JSON to S3
    s3 = boto3.client('s3', region_name=REGION_NAME, aws_access_key_id=AWS_ACCESS_KEY, aws_secret_access_key=AWS_SECRET_KEY)

    try:
        file_object = s3.get_object(Bucket=BUCKET_NAME, Key=FILE_NAME)
        # Read the Body stream of the S3 object and load the JSON
        file_content = json.load(file_object['Body'])

        return file_content, 200
    except ClientError as e:
        error_code = int(e.response['Error']['Code'])
        if error_code == 404:
            return { 'error': 'File not found in S3 bucket' }, 404
        else:
            return { 'error': 'Unknown error' }, 500


if __name__ == '__main__':
    app.run(debug=True, host='0.0.0.0')

File: samples/nodejs/AWS S3/main.js
const express = require('express');
const { S3Client, PutObjectCommand, GetObjectCommand } = require("@aws-sdk/client-s3");
const fs = require('fs').promises;
const { createReadStream } = require('fs');

const app = express();
app.use(express.json());

const REGION_NAME = 'us-west-2';
const BUCKET_NAME = 'my-sample-bucket';
const FILE_NAME   = 'file1.json';

const s3 = new S3Client({
  region: REGION_NAME,
  credentials: {
    accessKeyId: process.env.AWS_ACCESS_KEY,
    secretAccessKey: process.env.AWS_SECRET_KEY
  }
});

app.get('/', (req, res) => {
    res.json({ status: 'ok' });
});

app.post('/upload', async (req, res) => {
    let data = req.body;

    try {
        await fs.writeFile(FILE_NAME, JSON.stringify(data));
        const fileStream = createReadStream(FILE_NAME);

        const uploadParams = {
            Bucket: BUCKET_NAME,
            Key: FILE_NAME,
            Body: fileStream
        };
        await s3.send(new PutObjectCommand(uploadParams));
        res.json({ status: 'ok' });
    } catch (err) {
        res.status(500).json({ error: err.message });
    }
});

app.get('/download', async (req, res) => {
    const downloadParams = {
        Bucket: BUCKET_NAME,
        Key: FILE_NAME,
    };

    try {
        const { Body } = await s3.send(new GetObjectCommand(downloadParams));
        let responseBody = '';
        for await (let chunk of Body) {
            responseBody += chunk;
        }
        res.json(JSON.parse(responseBody));
    } catch (err) {
        if (err.name === 'NoSuchKey') {
            res.status(404).json({ error: 'File not found in S3 bucket' });
        } else {
            res.status(500).json({ error: 'Unknown error' });
        }
    }
});

app.listen(3000, () => console.log('App listening on port 3000'));
